projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -12,12 +12,89 @@ export function cloneArray(array, deep?: boolean) {
     }
     return arr;
 }
+
 /**
+ * Deep clones all first level keys of Obj2 and merges them to Obj1
+ * @param obj1 Object to merge into
+ * @param obj2 Object to merge from
+ * @returns Obj1 with merged cloned keys from Obj2
+ * @hidden
+ */
+export function mergeObjects(obj1: {}, obj2: {}) {
+    if (obj1 === null || obj1 === undefined) {
```

Comment:
If this is to remain somewhat comparable to `Object.assign` the first param is required, otherwise code might as well use clone directly